src/LanguageServer.ts

```diff
@@ -924,10 +925,9 @@ export class LanguageServer {
                 .filter(change => change.type === FileChangeType.Created)
                 //keep only the directories
                 .filter(change => util.isDirectorySync(change.srcPath));
-
-            //remove the created directories from the changes array (we will add back each of their files next)
-            changes = changes.filter(x => !directoryChanges.includes(x));
-
+            const stagingDirs = projects.map(p => p.builder?.options?.stagingDir ?? '').filter(stagingDir => stagingDir);
+            stagingDirs.push('.roku-deploy-staging');
+            stagingDirs.push('node_modules');
```

Comment:
I'm not sure this is actually safe. There are valid use cases for doing things like this in a bsconfig.json files array:

```{
    "files": [{
          "src": "../.roku-deploy-staging/someFile.brs",
           "dest": "source/someFile.brs"
       }, {
           "src": "node_modules/some-project/someLib.brs",
           "dest": "source/someLib.brs"
        }]
}
```

There's also the concept of standalone files (i.e. a file that's not represented by a bsconfig.json files array). So if you open the staging folder, it should spin up a new project for that file on its own and provide validation and editor help.